### PR TITLE
Update PropertyTabAttribute.Equals() nullability

### LINKED ISSUE #55178: PropertyTabAttribute.Equals(PropertyTabAttribute other) should likely accept null
Found during adding nullable annotations to TypeConverter (https://github.com/dotnet/runtime/pull/54961).

Most likely the signature for this method should look like:

```csharp
public bool Equals([NotNullWhen(true)] PropertyTabAttribute? other)
```

but currently it will throw NRE as it accesses members without any null checks

### PR DESCRIPTION
**Pull request by hrrrrustic**
Close #55178

### CODE REVIEW

File: src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/PropertyTabAttribute.cs [hunk 1]
```diff
@@ -145,34 +145,25 @@ private void InitializeTabClasses()
...
+            => Equals(other as PropertyTabAttribute);
 
-        public bool Equals(PropertyTabAttribute other)
+        public bool Equals([NotNullWhen(true)] PropertyTabAttribute? other)
         {
-            if (other == (object)this)
-            {
+            if (other is null)
+                return false;
+
```
Comment (buyaa-n) [inline src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/PropertyTabAttribute.cs:154]: NIT: Isn't this null check redundant? `ReferenceEquals` allows nulls
```suggestion
```

Comment (hrrrrustic) [reply src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/PropertyTabAttribute.cs:154]: But if ReferenceEquals returns false we'll have NRE in the next if
I can swap null-check and ReferenceEquals, but does it really matter?

**Review by buyaa-n — APPROVED** [review]
LGTM, thanks!

**Comment by buyaa-n** [comment]
Failures unrelated and reported

**Pull request merged**